cmd/run/logger/logger.go

```diff
@@ -0,0 +1,91 @@
+package logger
+
+import (
+	"bufio"
+	"io"
+	"sync"
+)
+
+var Enabled bool
+
+var bufferSize = 4096
+
+type Logger struct {
+	wmu sync.Mutex
+	pr  *io.PipeReader
+	pw  *io.PipeWriter
+
+	rmu     sync.RWMutex
+	lines   []string
+	current int
+}
+
+func New() *Logger {
+	pr, pw := io.Pipe()
+	return &Logger{
+		lines:   make([]string, bufferSize),
+		current: 0,
+
+		pr: pr,
+		pw: pw,
+	}
+}
+
+func (l *Logger) Loop() {
+	s := bufio.NewScanner(l.pr)
+	defer l.pr.Close()
+	defer l.pw.Close()
```

Comment:
the read loop is only responsible for the reader, so it shouldn't touch the writer